### PR TITLE
Use mapbox for geocoder

### LINKED ISSUE #570: Cache geocoder results
I just needed to set this up for a different project that hit the Google geocode API limit.
We don't make super heavy use of maps but maybe it's interesting to some to try it out?
I followed this guide https://github.com/codeforamerica/ohana-api/wiki/Customizing-the-geocoding-configuration but it's very Heroku specific (other project runs on Heroku) but maybe we can get it to run on Uberspace as well.

### PR DESCRIPTION
**Pull request by vsmart**
Closes #570 

todo:
* [x] switch geocoder provider to mapbox
* [x] fix geocoder tests and test fixtures
* [x] add a way to store the api key securely and privately

to deploy: 
* [ ] share mapbox access with folks
* [ ] merge
* [ ] put key on server 
* [ ] git push production

### CODE REVIEW

File: app/models/group.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ class Group < ActiveRecord::Base
   validates :country, presence: true
 
   mount_uploader :picture, PictureUploader
-  geocoded_by :location
+  geocoded_by :location, :lookup => :mapbox
```
Comment (til) [inline app/models/group.rb]: I know we are super inconsistent throughout our codebase with this, but let me nevertheless suggest to use the new Ruby hash syntax, at least for newly added code, ie `lookup: :mapbox`

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,4 @@ pickle-email-*.html
 .project
 .DS_Store
 config/secrets.yml
+.auth
```
Comment (vsmart) [inline .gitignore]: this should be removed

**Comment by sareg0** [comment]
@vsmart I haven't done the part about sharing mapbox access with the people. Is that something you were planning on doing?

**Pull request merged**